package/MDAnalysis/lib/distances.py

```diff
@@ -525,64 +523,134 @@ def _determine_method(reference, configuration, max_cutoff, min_cutoff=None,
+    """Capped distance evaluations using a KDtree method.
 
-    Uses minimum image convention if *box* is specified
+    Computes and returns an array containing pairs of indices corresponding to
+    entries in the `reference` and `configuration` arrays which are separated by
+    a distance lying within the specified cutoff(s). Employs a (periodic) KDtree
+    algorithm to find relevant distances.
 
-    Returns:
-    --------
-    pairs : array
-        Array of atom indices which are within the specified cutoff distance.
-        Every pairs `(i, j)` corresponds to i-th particle in reference and
-        j-th particle in configuration
-    distance : (optional) array
-        Distance between two atoms corresponding to the (i, j) indices
-        in pairs. Only returns when ``return_distances``
-         is set to ``True``
+    Optionally, these distances can be returned as well.
+
+    If the optional argument `box` is supplied, the minimum image convention is
+    applied when calculating distances. Either orthogonal or triclinic boxes are
+    supported.
+
+    Parameters
+    ----------
+    reference : numpy.ndarray
+        Reference coordinate array with shape ``(3,)`` or ``(n, 3)`` (dtype will
+        be converted to ``numpy.float32`` internally).
+    configuration : array
+        Configuration coordinate array with shape ``(3,)`` or ``(m, 3)`` (dtype
+        will be converted to ``numpy.float32`` internally).
+    max_cutoff : float
+        Maximum cutoff distance between `reference` and `configuration`
+        coordinates.
+    min_cutoff : float, optional
+        Minimum cutoff distance between `reference` and `configuration`
+        coordinates.
+    box : numpy.ndarray, optional
+        The unitcell dimensions of the system, which can be orthogonal or
+        triclinic and must be provided in the same format as returned by
+        :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:\n
+        ``[lx, ly, lz, alpha, beta, gamma]``.
+    return_distances : bool, optional
+        If set to ``True``, distances will also be returned.
 
+    Returns
+    -------
+    pairs : numpy.ndarray
+        Pairs of indices, corresponding to coordinates in the `reference` and
+        `configuration` arrays such that the distance between them lies within
+        the interval (`min_cutoff`, `max_cutoff`].
+        Each row in `pairs` is an index pair ``[i, j]`` corresponding to the
+        ``i``-th coordinate in `reference` and the ``j``-th coordinate in
+        `configuration`.
+    distances : numpy.ndarray, optional
+        Distances corresponding to each pair of indices. Only returned if
+        `return_distances` is ``True``. ``distances[k]`` corresponds to the
+        ``k``-th pair returned in `pairs` and gives the distance between the
+        coordinates ``reference[pairs[k, 0]]`` and
+        ``configuration[pairs[k, 1]]``.
     """
     from .pkdtree import PeriodicKDTree
```

Comment:
We rather have all imports at the top of the file instead of methods. If it has to be inside a method (eg to avoid circular imports) then it should be commented. See https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide#importing-modules

Reply:
(Yes, I know, this is not your code...)

Reply:
I'll move the imports to the top if they don't have any side effects (which they shouldn't).